lddwrap/__init__.py

```diff
@@ -137,51 +114,53 @@ def _parse_line(line: str) -> Optional[Dependency]:
     # with soname but not found: 'libboost_program_options.so.1.62.0 => not found'
     # with soname but without rpath: 'linux-vdso.so.1 =>  (0x00007ffd7c7fd000)'
     # pylint: enable=line-too-long
+    soname = None
+    dep_path = None
+    mem_address = None
     if '=>' in line:
-        if len(parts) != 4:
-            raise RuntimeError(
-                "Expected 4 parts in the line but found {}: {}".format(
-                    len(parts), line))
-
-        soname = None
-        dep_path = None
-        mem_address = None
+        mtch = _LDD_ARROW_OUTPUT_RE.match(line)
+        if not mtch:
+            raise RuntimeError(("Unexpected ldd output. Expect a regex match {}, "
+                                "but got: {!r}").format(_LDD_ARROW_OUTPUT_RE.pattern,
+                                                        line))
         if found:
-            soname = parts[0]
-            if parts[2] != '':
-                dep_path = pathlib.Path(parts[2])
-
-            mem_address = _strip_mem_address(text=parts[3])
+            soname = mtch["soname"]
+            if mtch["dep_path"]:
+                dep_path = pathlib.Path(mtch["dep_path"])
+            if mtch["mem_address"]:
+                mem_address = mtch["mem_address"]
         else:
             if "/" in parts[0]:
```

Comment:
If I understood your change correctly, you probably want to delete the line 104 where `parts` is set and not use `parts[0]` here at all?

Reply:
Oh yeah, this was my mistake. I actually wanted to delete the `parts` variable but completely forgot about it.